bitnami/common/templates/_affinities.tpl

```diff
@@ -77,15 +78,29 @@ preferredDuringSchedulingIgnoredDuringExecution:
           {{- end }}
       topologyKey: {{ include "common.affinities.topologyKey" (dict "topologyKey" .topologyKey) }}
     weight: 1
+  {{- range $extraPodAffinityTerms }}
+  - podAffinityTerm:
+      labelSelector:
+        matchLabels: {{- (include "common.labels.matchLabels" $.context) | nindent 10 }}
+          {{- if not (empty $component) }}
+          {{ printf "app.kubernetes.io/component: %s" $component }}
+          {{- end }}
+          {{- range $key, $value := .extraMatchLabels }}
+          {{ $key }}: {{ $value | quote }}
+          {{- end }}
+      topologyKey: {{ include "common.affinities.topologyKey" (dict "topologyKey" .topologyKey) }}
+    weight: {{ .weight | default 1 -}}
+  {{- end -}}
```

Comment:
The "extra" pod affinity terms should be different, right? I mean, it looks like we're adding new ones using the same weight, topologyKey, labelSelector, etc.

Reply:
In our previous discussion I gave an example of PodAffinity I'm aiming to generate with with this changes. It allows to define topologyKey and weight for each extra podAffinityTerm. Also we can define additional labels for every extraPodAffinityTerm by using  extraMatchLabels. For example:
```
 podAntiAffinity: {{- include "common.affinities.pods" (dict "type" $.Values.podAntiAffinityPreset "component" "clickhouse" "extraPodAffinityTerms" (list (dict "extraMatchLabels" (dict "shard" $i) "topologyKey" "topology.kubernetes.io/zone" "weight" 10) (dict "extraMatchLabels" (dict "FOO" "BAR") "topologyKey" "topology.kubernetes.io/region" "weight" 15)) "context" $) | nindent 10 }}

 ```
`extraPodAffinityTerms` is a list full of dicts, each dict describes different podAffinityTerm individually, the only common thing is a set of common labels.
With `.Values.podAntiAffinityPreset` set to soft, template will generate affinity like this:
```
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
            - podAffinityTerm:
                labelSelector:
                  matchLabels:
                    app.kubernetes.io/name: clickhouse
                    app.kubernetes.io/instance: clickhouse-bitnami
                    app.kubernetes.io/component: clickhouse
                topologyKey: kubernetes.io/hostname
              weight: 1
            - podAffinityTerm:
                labelSelector:
                  matchLabels:
                    app.kubernetes.io/name: clickhouse
                    app.kubernetes.io/instance: clickhouse-bitnami
                    app.kubernetes.io/component: clickhouse
                    shard: "0"
                topologyKey: topology.kubernetes.io/zone
              weight: 10
            - podAffinityTerm:
                labelSelector:
                  matchLabels:
                    app.kubernetes.io/name: clickhouse
                    app.kubernetes.io/instance: clickhouse-bitnami
                    app.kubernetes.io/component: clickhouse
                    FOO: "BAR"
                topologyKey: topology.kubernetes.io/region
              weight: 15

```

Reply:
Oh! Thanks, I get it now.